sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs

```diff
@@ -280,6 +306,41 @@ public async Task StartRecognizeCustomFormsLabeled(bool useStream)
             Assert.AreEqual("948284", form.Fields[name].ValueText.Text);
         }
 
+        /// <summary>
+        /// Verifies that the <see cref="FormRecognizerClient" /> is able to connect to the Form
+        /// Recognizer cognitive service and perform operations.
+        /// </summary>
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task StartRecognizeCustomFormsFromUriThrowsForNonExistingContent(bool labeled)
+        {
+            var client = CreateInstrumentedClient();
+            var invalidUri = new Uri("https://idont.ex.ist");
+            var modelId = await GetModelIdAsync(labeled);
+
+            var operation = await client.StartRecognizeCustomFormsFromUriAsync(modelId, invalidUri);
+            RequestFailedException capturedException = default;
+
+            try
+            {
+                await operation.WaitForCompletionAsync();
+            }
+            catch (RequestFailedException ex)
+            {
+                capturedException = ex;
+            }
+
+            string expectedErrorCode = labeled ? "3003" : "2003";
+
+            Assert.NotNull(capturedException);
+            Assert.AreEqual(expectedErrorCode, capturedException.ErrorCode);
+
+            Assert.True(operation.HasCompleted);
+            Assert.True(operation.HasValue);
```

Comment:
I'm wondering if HasValue should throw since we didn't get back ReadResult/PageResult/DocumentResult values.

Reply:
Do you mean `Value` should throw? I think it would be reasonable to have `Value` throwing and `HasValue` returning `false`, like a running operation.

Reply:
Yes, that's what I meant - thanks for the correction!